GitCommands/Git/GitItemStatus.cs

```diff
@@ -46,7 +47,13 @@ private enum Flags
 
         private Flags _flags;
 
-        public string? Name { get; set; }
+        public GitItemStatus(string name)
+        {
+            Requires.NotNull(name, nameof(name));
+            Name = name;
+        }
+
+        public string Name { get; set; }
```

Comment:
Here is the example I called out in the PR description, of introducing a constructor to avoid marking a property as nullable.